tasks/pipeline.py

```diff
@@ -463,6 +465,99 @@ def trigger_child_pipeline(_, git_ref, project_name, variables="", follow=True):
         print("Child pipeline finished successfully")
 
 
+def parse(commit_str):
+    lines = commit_str.split("\n")
+    title = lines[0]
+    url = "NO_URL"
+    pr_id_match = re.search(r".*\(#(\d+)\)", title)
+    if pr_id_match is not None:
+        url = f"https://github.com//DataDog/datadog-agent/pull/{pr_id_match.group(1)}"
+    author = lines[1]
+    author_email = lines[2]
+    files = lines[3:]
+    return title, author, author_email, files, url
+
+
+def is_system_probe(owners, files):
+    target = {
+        ("TEAM", "@DataDog/Networks"),
+        ("TEAM", "@DataDog/universal-service-monitoring"),
+        ("TEAM", "@DataDog/ebpf-platform"),
+        ("TEAM", "@DataDog/agent-security"),
+    }
+    for f in files:
+        match_teams = set(owners.of(f)) & target
+        if len(match_teams) != 0:
+            return True
+
+    return False
+
+
+@task
+def changelog(ctx, new_commit_sha):
+    old_commit_sha = ctx.run(
+        "aws ssm get-parameter --region us-east-1 --name "
+        "ci.datadog-agent.gitlab_changelog_commit_sha --with-decryption --query "
+        "\"Parameter.Value\" --out text",
+        hide=False,
+    ).stdout.strip()
+    print(f"Generating changelog for commit range {old_commit_sha} to {new_commit_sha}")
+    commits = ctx.run(f"git log {old_commit_sha}..{new_commit_sha} --pretty=format:%h", hide=True).stdout.split("\n")
+    owners = read_owners(".github/CODEOWNERS")
+    messages = []
+    unique_emails = set()
+
+    for commit in commits:
+        # see https://git-scm.com/docs/pretty-formats for format string
+        commit_str = ctx.run(f"git show --name-only --pretty=format:%s%n%aN%n%aE {commit}", hide=True).stdout
+        title, author, author_email, files, url = parse(commit_str)
+        if is_system_probe(owners, files):
+            message = f"{title} (<{url}|PR Link>) {author}"
+            messages.append(message)
+            if "dependabot" not in author_email and "github-actions" not in author_email:
+                unique_emails.add(author_email)
+
+    with open("system_probe_commits.txt", "w") as file:
+        content = (
+            f"Changelog for commit range: `{old_commit_sha}` to `{new_commit_sha}`\n"
+            + "\n".join(messages)
+            + "\n:wave: Authors, please check relevant "
+            "<https://ddstaging.datadoghq.com/dashboard/kfn-zy2-t98|dashboards> for issues:"
+        )
+        file.write(content)
+
+    with open("unique_emails.txt", "w") as file:
+        file.write("\n".join(unique_emails))
+
+
+@task
+def post_changelog(ctx, new_git_sha):
+    results = []
+    with open('unique_emails.txt', 'r') as email_file:
+        for email in email_file:
+            result = ctx.run(f"email2slackid ${email.strip()}", hide=True)
```

Comment:
It looks like can you pass multiple emails to `email2slackid`

Reply:
rewriting this to take advantage of this will likely speed up runtime considerably, assuming `email2slackid` is hitting the slack ID in bulk in that case